### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by pelzolga123**

### CODE REVIEW

File: app/controllers/sessions_controller.rb [hunk 1]
```diff
@@ -0,0 +1,22 @@
+class SessionsController < ApplicationController
+  def new
+    @user = User.new
+  end
+
+  def create
+    user = User.find_by(email: params[:session][:email].downcase)
```
Comment (SunnyAsar) [inline app/controllers/sessions_controller.rb]: - opps! `signin` link is broken
Screenshot 2019-09-04 at 12 37 49 AM

File: app/controllers/posts_controller.rb [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    @posts = Post.all
+  end
+
+  def new
+    @post = Post.new
+  end
+
+  def create
+    @post = Post.new(post_params)
+    @post.user_id = current_user.id
```
Comment (SunnyAsar) [inline app/controllers/posts_controller.rb]: let's use the relationship we have, a better way would be 
`@post = current_user.posts.build(post_params)`

File: app/controllers/posts_controller.rb [hunk 3]
```diff
@@ -0,0 +1,30 @@
+class PostsController < ApplicationController
```
Comment (SunnyAsar) [inline app/controllers/posts_controller.rb]: add this comment `# frozen_string_literal: true` at the top of your classes to get rid of the linting error

Comment (tundeiness) [reply app/controllers/posts_controller.rb]: SunnyAsar you should know better that this is Rubocop linting (# frozen_string_literal: true). This thing you pointed out is not really significant..you want to know why? all these will not be needed in production.

File: app/controllers/posts_controller.rb [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+  end
+
+  def new
+    @post = Post.new
+  end
+
+  def create
+    @post = current_user.posts.build(post_params)
+    @post.user_id = current_user.id
+    @post.save
```
Comment (mosaaleb) [inline app/controllers/posts_controller.rb]: - [ ] As pointed out before, `@post = current_user.posts.build(post_params)` has created post through associations. So we don't really need `@post.user_id = curren_user.id`

File: config/routes.rb [hunk 5]
```diff
@@ -1,3 +1,12 @@
+# frozen_string_literal: true
+
 Rails.application.routes.draw do
-  # For details on the DSL available within this file, see http://guides.rubyonrails.org/routing.html
+  # For details on the DSL available within this file, see
+  # http://guides.rubyonrails.org/routing.html
+  resources :sessions, only: [:new, :create, :destroy]
+  match '/signin',  to: 'sessions#new',         via: 'get'
+  match '/signout', to: 'sessions#destroy',     via: 'delete'
```
Comment (mosaaleb) [inline config/routes.rb]: - **[OPTIONAL]** Rails > 4 provide a shorthand way of expessing constraint by replacing `match`  with the desired HTTP method (get, post, patch, etc). Also a common shorhand for defining a route is to use `=>` operator. So we can use `get '/signin' => 'sessions#destroy'` and same for `signout`. [1](https://guides.rubyonrails.org/routing.html#connecting-urls-to-code)

File: config/routes.rb [hunk 6]
```diff
@@ -1,3 +1,12 @@
+# frozen_string_literal: true
+
 Rails.application.routes.draw do
-  # For details on the DSL available within this file, see http://guides.rubyonrails.org/routing.html
+  # For details on the DSL available within this file, see
+  # http://guides.rubyonrails.org/routing.html
+  resources :sessions, only: [:new, :create, :destroy]
```
Comment (mosaaleb) [inline config/routes.rb]: - [ ] There are two ways of defining routes either [Resourceful Routes](https://guides.rubyonrails.org/routing.html#resource-routing-the-rails-default) or [Non-Resourceful Routes](https://guides.rubyonrails.org/routing.html#non-resourceful-routes). You have already used both of them for sessions, while one way is only enough. If you choose to go with the `Non-Resourceful Routes`, then we will still need a route for creating session (POST) with 
`post '/signin', to: 'sessions#create'`

**Pull request merged**